conan/tools/gnu/autotoolsdeps.py

```diff
@@ -1,91 +1,63 @@
 from conan.tools.env import Environment
-from conans.model.build_info import DepCppInfo
+from conan.tools.gnu.autotoolsdeps_flags import AutoToolsDepsFlags
+from conans.model.new_build_info import NewCppInfo
 
 
 class AutotoolsDeps:
     def __init__(self, conanfile):
         # Set the generic objects before mapping to env vars to let the user
         # alter some value
         self._conanfile = conanfile
-
-        self.libs = []
-        self.system_libs = []
-        self.include_paths = []
-        self.lib_paths = []
-        self.defines = []
-        self.cflags = []
-        self.cxxflags = []
-        self.sharedlinkflags = []
-        self.exelinkflags = []
-        self.frameworks = []
-        self.framework_paths = []
-        self.sysroot = None
-
-        def merge_lists(seq1, seq2):
-            return [s for s in seq1 if s not in seq2] + seq2
-
-        def merge(dep):
-            dep_cpp_info = DepCppInfo(dep.cpp_info)  # To deal with components
-            self.system_libs = merge_lists(self.system_libs, dep_cpp_info.system_libs)
-            self.include_paths = merge_lists(self.include_paths, dep_cpp_info.include_paths)
-            self.lib_paths = merge_lists(self.lib_paths, dep_cpp_info.lib_paths)
-            self.framework_paths = merge_lists(self.framework_paths, dep_cpp_info.framework_paths)
-            self.libs = merge_lists(self.libs, dep_cpp_info.libs)
-            self.frameworks = merge_lists(self.frameworks, dep_cpp_info.frameworks)
-
-            # Note these are in reverse order
-            self.defines = merge_lists(dep_cpp_info.defines, self.defines)
-            self.cxxflags = merge_lists(dep_cpp_info.cxxflags, self.cxxflags)
-            self.cflags = merge_lists(dep_cpp_info.cflags, self.cflags)
-            self.sharedlinkflags = merge_lists(dep_cpp_info.sharedlinkflags, self.sharedlinkflags)
-            self.exelinkflags = merge_lists(dep_cpp_info.exelinkflags, self.exelinkflags)
-
-            if not self.sysroot:
-                self.sysroot = dep_cpp_info.sysroot
-
-        for d in self._conanfile.dependencies.host.values():
-            merge(d)
+        self._cpp_info = None
+
+    @property
+    def cpp_info(self):
+        if self._cpp_info is None:
+            self._cpp_info = NewCppInfo()
```

Comment:
Only CMake and now Autotools use the new cpp info. We should plan move the rest of them.